app/lib/dns-server.ts

```diff
@@ -0,0 +1,239 @@
+import {
+  CreateHostedZoneCommand,
+  ChangeResourceRecordSetsCommand,
+  GetChangeCommand,
+  ListResourceRecordSetsCommand,
+  ResourceRecordSet,
+} from "@aws-sdk/client-route-53";
+import { route53Client } from "./route53Client";
+
+export const createHostedZone = async (domain: string) => {
+  try {
+    const command = new CreateHostedZoneCommand({
+      Name: domain,
+      CallerReference: new Date().toString(),
+    });
+    const response = await route53Client.send(command);
+
+    if (!response.HostedZone?.Id) {
+      throw new Error("Could create hosted zone upon request");
+    }
+
+    return response.HostedZone?.Id.replace("/hostedzone/", "");
+  } catch (error) {
+    throw new Error(`Error while creating hosted zone, ${error}`);
+  }
+};
+
+export const resetHostedZone = async (hostedZoneId: string) => {
+  try {
+    const resourceRecordSets = await listRecords(hostedZoneId);
+
+    resourceRecordSets.forEach(async (resourceRecordSet) => {
+      if (resourceRecordSet.Type === RecordType.A) {
+        await deleteRecord(
+          RecordType.A,
+          resourceRecordSet.Name!,
+          resourceRecordSet.ResourceRecords![0].Value!,
+          hostedZoneId
+        );
+      } else if (resourceRecordSet.Type === RecordType.AAAA) {
+        await deleteRecord(
+          RecordType.AAAA,
+          resourceRecordSet.Name!,
+          resourceRecordSet.ResourceRecords![0].Value!,
+          hostedZoneId
+        );
+      } else if (resourceRecordSet.Type === RecordType.CNAME) {
+        await deleteRecord(
+          RecordType.CNAME,
+          resourceRecordSet.Name!,
+          resourceRecordSet.ResourceRecords![0].Value!,
+          hostedZoneId
+        );
+      } else if (resourceRecordSet.Type === RecordType.TXT) {
+        await deleteRecord(
+          RecordType.TXT,
+          resourceRecordSet.Name!,
+          resourceRecordSet.ResourceRecords![0].Value!,
+          hostedZoneId
+        );
+      }
+    });
+  } catch (error) {
+    throw new Error(`Error while deleting hosted zone: ${error}`);
+  }
+};
+
+export const createRecord = async (
+  type: RecordType,
+  name: string,
+  value: string,
```

Comment:
We need to do some validation on the value as well.  Depending on the record type, we'll expect different things:

If the `RecordType` is `A` (ipv4) or `AAAA` (ipv6) we could use https://www.npmjs.com/package/is-ip to check it.

If it's a CNAME, we should probably let Route53 decide if it's valid (there's no standard definition of what this should be afaik).  A string of length >= 1 is probably enough

For txt, we could again take any value that is length >= 1

cc @dadolhay for thoughts as well.

Reply:
I thought there would be another interface to validate the input. But I would love to add the validation. Will be added.

Reply:
We can validate in the front-end too, but this adds some extra ability to debug faulty data.